langchain4j-google-ai-gemini/src/main/java/dev/langchain4j/model/googleai/GoogleAiEmbeddingModel.java

```diff
@@ -0,0 +1,193 @@
+
+        for (int i = 0; i < numberOfBatches; i++) {
+            int startIndex = MAX_NUMBER_OF_SEGMENTS_PER_BATCH * i;
+            int lastIndex = Math.min(startIndex + MAX_NUMBER_OF_SEGMENTS_PER_BATCH, numberOfEmbeddings);
+
+            if (startIndex >= numberOfEmbeddings) break;
+
+            GoogleAiBatchEmbeddingRequest batchEmbeddingRequest = new GoogleAiBatchEmbeddingRequest();
+            batchEmbeddingRequest.setRequests(embeddingRequests.subList(startIndex, lastIndex));
+
+            Call<GoogleAiBatchEmbeddingResponse> geminiBatchEmbeddingResponseCall =
+                withRetry(() -> this.geminiService.batchEmbed(this.modelName, this.apiKey, batchEmbeddingRequest));
+
+            GoogleAiBatchEmbeddingResponse geminiResponse;
+            try {
+                retrofit2.Response<GoogleAiBatchEmbeddingResponse> executed = geminiBatchEmbeddingResponseCall.execute();
+                geminiResponse = executed.body();
+
+                if (executed.code() >= 300) {
+                    try (ResponseBody errorBody = executed.errorBody()) {
+                        GeminiError error = GSON.fromJson(errorBody.string(), GeminiErrorContainer.class).getError();
+
+                        throw new RuntimeException(
+                            String.format("%s (code %d) %s", error.getStatus(), error.getCode(), error.getMessage()));
+                    }
+                }
+            } catch (IOException e) {
+                throw new RuntimeException("An error occurred when calling the Gemini API endpoint (embedAll).", e);
+            }
+
+            if (geminiResponse != null) {
+                allEmbeddings.addAll(geminiResponse.getEmbeddings().stream()
+                    .map(values -> Embedding.from(values.getValues()))
+                    .collect(Collectors.toList()));
+            } else {
+                throw new RuntimeException("Gemini embedding response was null (embedAll)");
+            }
+        }
+
+        return Response.from(allEmbeddings);
+    }
+
+    private GoogleAiEmbeddingRequest getGoogleAiEmbeddingRequest(TextSegment textSegment) {
+        GeminiPart geminiPart = GeminiPart.builder()
+            .text(textSegment.text())
+            .build();
+
+        GeminiContent content = new GeminiContent(Collections.singletonList(geminiPart), null);
+
+        GoogleAiEmbeddingTaskType taskType = null;
+        if (textSegment.metadata() != null && textSegment.metadata().getString("taskType") != null) {
+            taskType = GoogleAiEmbeddingTaskType.valueOf(textSegment.metadata().getString("taskType"));
+        } else if (this.taskType != null) {
+            taskType = this.taskType;
+        }
+
+        String title;
+        if (textSegment.metadata() != null && textSegment.metadata().getString("title") != null) {
+            title = textSegment.metadata().getString("title");
+        } else {
+            title = this.titleMetadataKey;
```

Comment:
```suggestion
            title = textSegment.metadata().getString(titleMetadataKey);
```

It seems it was supposed to take it from the metadata? Then this logic can be simplified: `titleMetadataKey` can have default vlaue "title" assigned in ctor.

Reply:
I used the same logic as for the task type.
First look if there's a title on a per segment basis, if not, check if there's one defined in the constructor, otherwise null.
That said, I think I'll add just one surrounding `if` because the title only makes sense (for Google's embedding model) only when the task type is `RETRIEVAL_DOCUMENT`.

Reply:
Hmm, does it make sense to have the same title for many segments? Is there a use case for this?

Reply:
Using the same title was clearly a mistake. Thanks for spotting it.